lighthouse-core/runner.js

```diff
@@ -325,7 +325,12 @@ class Runner {
         ...sharedAuditContext,
       };
 
-      const product = await audit.audit(artifacts, auditContext);
+      const requiredArtifacts = audit.meta.requiredArtifacts
+        .reduce((requiredArtifacts, artifactName) => {
+          requiredArtifacts[artifactName] = artifacts[artifactName];
+          return requiredArtifacts;
+        }, /** @type {LH.Artifacts} */ ({}));
```

Comment:
this is unfortunate, but fine enough. Maybe add a line about the masquerading type to the above comment as well?

I was hoping at one point that we could type the `artifacts` input of an audit based on its declared `requiredArtifacts`, but it would require not having `Array<string>` for the `requiredArtifacts` type, was finicky because it was nested on the `meta` getter, and didn't gain us all that much, but maybe in some future version of typescript...

Reply:
yeah I thought about it for about 5 seconds before thinking there was no way it was going to be usable 😆 

hopefully one day typescript will let you work your magic here :)